### ISSUE TITLE
Unexpected Bigwig Output

### ISSUE THREAD

**Issue by CriticalSci** [ncbi/BAMscale#14]
Hi there,
I tried running `BAMscale scale --operation subtract` on two ChIPseq files, it processed them swiftly however the resulting bigwig files are causing trouble with downstream tools such as deeptools' plotProfile.
It also appears as bars in the IGV browser rather than the usual signal track as shown:
<img width="413" alt="Screenshot 2021-02-17 at 15 58 17" src="https://user-images.githubusercontent.com/57266625/108173359-0079f800-7139-11eb-9630-31e821fb9ce2.png">

Converting to wig format using bigWigToWig seems to indicate the bigwigs are fixedStep, could this be the problem?
Any option to output to variableStep bigwig?

**Comment by pongorlorinc** [comment]
Hi CriticalSci,

Thanks for reaching out! 

Using fixedStep and variableStep (to my knowledge) is not really different, just the block sizes used differ to store the data. FixedStep will not have any start coordinate, while variablestep does (making the data block slightly larger).

In both cases, multiple bases are binned together into one bar. What determines how smooth the track will be is the binSize, which refers to the number of bases that will be averaged to output a block (as can be seen in your figure). In case of BAMscale, 5 bases are binned together by default (this is pretty high resolution, usually too high for standard analyses), in IGV tools 25 bases are binned when creating tdf files (meaning that the bars will be 5x wider compared to BAMscale), while in deeptools  bamCoverage 50 bases are binned together (10x wider than BAMscale). Larger bin sizes are not a drawback though, as generally they retain most of the needed information (peaks or regions are at least a few hundred bases wide), and the output file size is greatly reduced.

To summarize, bigwigs are not intended to have single-base resolution to reduce file sizes. The binSize is different for each tool (but it can be altered if needed, in case of BAMscale it is the "--binsize <binsize>" or "-z <binsize>" flags). This is why, when you zoom in to a single-base resolution in the browser, you start seeing the bars.

What problems do you get with the deeptools plotProfile? I might try to reproduce it to see if there is any work-around.

Best wishes,
Lorinc

**Comment by CriticalSci** [comment]
Hi Lorinc,

Thanks for the swift reply!

Right, I just mentioned the formats as I noticed that the unpacked bigwigs generated by Deeptools (which plot correctly) look like bedgraphs or variable step formats originally.

Maybe Deeptools' plotProfile just doesn't handle fixedStep bigwigs properly?

I'm running computeMatrix and plotProfile/plotHeatmap from the deeptools package on bigwigs generated by BAMscale or deeptools' alternatives (bamCoverage+bigwigCompare; which are much slower).

The heatmaps look fairly normal but the profiles are missing data it seems (BAMscale on the left; deepTools on the right).
![Screenshot 2021-02-17 at 21 09 03](https://user-images.githubusercontent.com/57266625/108209394-3503a900-7165-11eb-8ea4-38784bc5a753.png)

Jean-Michel

**Comment by pongorlorinc** [comment]
Hi Jean-Michel!

This is so strange, that the profile is so different. The heatmap part is not surprisingly slightly different, as BAMscale has a different normalization, but the profile should be similar too.

In our experience, we didn't have any issues with the plotHeatmap functions when using data from BAMscale.

Would it be possible for you to share some of the data? (eg. the bigwig file and peak coordinates?) I would like to see if there are any strange values that might cause this problem.

Also, could you send a higher resolution of the profile on top? It is a bit pixely, and I can't see the axis.

Best wishes,
Lorinc
